sql/table.cc

```diff
@@ -8479,6 +8484,199 @@ LEX_CSTRING *fk_option_name(enum_fk_option opt)
   return names + opt;
 }
 
+TR_table::TR_table(THD* _thd, bool rw) : thd(_thd)
+{
+  static const LString table_name("transaction_registry");
+  init_one_table(LEX_STRING_WITH_LEN(MYSQL_SCHEMA_NAME),
+                 XSTRING_WITH_LEN(table_name), table_name, rw ? TL_WRITE : TL_READ);
+  open_tables_backup= new Open_tables_backup;
```

Comment:
I think have `open_tables_backup` as a value member is better than pointer: less writing, no possibility of allocation failure.

Reply:
Do you think it is worth to relocate `TR_table` to somewhere else? `Open_tables_backup` is inaccessible from `table.h` (declared in `sql_class.h`).

Reply:
Either move, or include `sql_class.h`. I don't think we care much about compilation speed. At least, as I can see from MariaDB code its writers do not care indeed :)

Reply:
It is really hard to do. `item_vers.h` requires `TR_table::field_id_t`. `item_vers.h` is included from `item.h:5114` from ... from `sql_class.h:498`. `Open_tables_backup` is declared at `sql_class.h:1532`. Rearranging `sql_class.h` is like disturbing a beehive! And I would not tear down `TR_table` and `field_id_t`. I don't think this allocation is so important.

Reply:
It's not worth it then.